[Rachel Cheah] ip

RachelCheah opened the pull request:
DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ *SUPER* FAST to use

All you need to do is,

1. download it from [here](https://github.com/RachelCheah/ip/releases/tag/A-Packages).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

RachelCheah committed:
Level-1

RachelCheah committed:
Level-2

RachelCheah committed:
Addded new Class Task and Level-3

RachelCheah committed:
Level-4

RachelCheah committed:
A-TextUiTesting

RachelCheah committed:
Level-5

RachelCheah committed:
Level-6

RachelCheah committed:
Level-7

RachelCheah committed:
Level-8

RachelCheah committed:
Exceptions Classes

RachelCheah committed:
Commands OOP

RachelCheah committed:
Tasks OOP

RachelCheah committed:
A-MoreOOP

RachelCheah committed:
A-Packages

qinguorui2001 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+
```
i recommend to list the imports from java... before those from duke..., since it follows check style.

qinguorui2001 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,50 @@
+        while (!isExit) {
+            try {
+                String fullCommand = ui.readCommand();
+                //#ui.showLine(); // show the divider line ("_______")
```
It's free to delete lines which are commented out to make the code clear, since you can still review the old version through source tree or git if u think it's better to change back.

qinguorui2001 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+                }
+            default:
+                throw new NoSuchCommandException("Invalid Command");
+        }
```
I think you can also use a try catch for the whole switch cases, instead of using try catch for each case for the purpose of making code short.

qinguorui2001 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,57 @@
+                }
+                return tasklist;
+            }
+
```
it could be better to be careful with the indentations.

qinguorui2001 commented on src/main/java/duke/Tasklist.java:
```diff
@@ -0,0 +1,69 @@
+    }
+
+
+
```
Similar problems as above, indentation between methods and javadoc need to be improved.

qinguorui2001 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,63 @@
+    }
+
+    public static Command parse(String input) throws NoSuchCommandException {
+        switch(input.split(" ")[0]) {
```
```suggestions
switch(...) {
case "list":
    ...
...
}
```
Be careful about this, for this module the requirement for writing switch cases is little different from what we encountered before.

qinguorui2001 reviewed (commented):
Generally, i think you did a good task, just be careful about some java coding standards.

nhjryan commented on src/main/java/duke/exceptions/NoSuchCommandException.java:
```diff
@@ -0,0 +1,7 @@
+package duke.exceptions;
+
+public class NoSuchCommandException extends Exception{
```
Great use of creating your own exception to deal with errors in the code. Perhaps instead of using `super(message);`, you could print "Invalid command" here and append any additional information after it, since I noticed that most of the messages specified were "Invalid command".

nhjryan reviewed (commented):
LGTM! The code adheres to the coding standard specified and is neat and readable. :)

RachelCheah committed:
A-JUnit

srj31 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,77 @@
+public class Task {
+
+    private final String name;
+    private boolean done;
```
Perhaps naming it isDone would be better, since it helps to understand later that the variable is boolean.

srj31 commented on src/main/java/duke/tasks/ToDosTask.java:
```diff
@@ -0,0 +1,29 @@
+        return input.split("todo")[1].strip();
+    }
+
+    public static void isLegitInput(String input) throws NoSuchCommandException {
```
Naming it checkInput would have been better, since isLegitInput would be better for a boolean function.

srj31 reviewed (commented):
LGTM. Adding JavaDoc comments would have made the code more readable. The coding standard seems to have been followed pretty well otherwise.

RachelCheah committed:
A-JavaDoc

RachelCheah committed:
A-CodingStandard

RachelCheah committed:
Level-9

RachelCheah committed:
Merge branch 'branch-A-JavaDoc' into branch-Level-9

* branch-A-JavaDoc:
  A-JavaDoc

# Conflicts:
#	src/main/java/duke/Parser.java

RachelCheah committed:
Merge Parser.java

RachelCheah committed:
Merge branch 'branch-A-CodingStandard' into branch-Level-9

* branch-A-CodingStandard:
  A-CodingStandard

# Conflicts:
#	src/main/java/duke/Parser.java

RachelCheah committed:
Fixed Indentation

RachelCheah committed:
Update README.md

RachelCheah committed:
A-Grade and CommandTest

RachelCheah committed:
A-CheckStyle

RachelCheah committed:
Setting up the GUI before adding logic

RachelCheah committed:
Level-10 GUI

RachelCheah committed:
Fixed Formatting

RachelCheah committed:
To generate JAR file

RachelCheah committed:
A-Assertions

RachelCheah committed:
A-CodeQuality

RachelCheah committed:
Merge pull request #1 from RachelCheah/branch-Assertions

A-Assertions

RachelCheah committed:
Merge branch 'branch-Assertions' into branch-A-CodeQuality

RachelCheah committed:
Merge pull request #2 from RachelCheah/branch-A-CodeQuality

A-CodeQuality

RachelCheah committed:
B-FixedDurationTasks and fixed delete/done text shown

Currently, delete/done command is saving the wrong task to temp

UI is showing the wrong task even though the delete/done works correctly

Changed the printing of task to show the correct task instead

RachelCheah committed:
Merge pull request #3 from RachelCheah/branch-BCD-Extension

B-FixedDurationTasks and fixed delete/done text shown

Currently, delete/done command is saving the wrong task to temp

UI is showing the wrong task even though the delete/done works correctly

Changed the printing of task to show the correct task instead

RachelCheah committed:
Dummy Commit to push Branch

RachelCheah committed:
Dummy Commit for Branch Level 8

RachelCheah committed:
Dummy Commit for Branch Level 10

RachelCheah committed:
Dummy Commit for Level 8

RachelCheah committed:
Merge pull request #5 from RachelCheah/branch-Level-8

Dummy Commit for Level 8

RachelCheah committed:
Updated User Guide

RachelCheah committed:
Merge branch 'branch-Level-10'

RachelCheah committed:
Merge remote-tracking branch 'origin/branch-Level-10'

# Conflicts:
#	src/main/resources/view/DialogBox.fxml

RachelCheah committed:
Dummy Commit for Branch 7

RachelCheah committed:
Merge branch 'branch-Level-7'

RachelCheah committed:
Update README.md

Minor Changes to README

RachelCheah committed:
Updated Tasks to Read Time and updated README

RachelCheah committed:
Updated Ui.png

RachelCheah committed:
Update README.md
